Wabbajack.VirtualFileSystem/Context.cs

```diff
@@ -39,85 +39,99 @@ static Context()
 
         public StatusUpdateTracker UpdateTracker { get; set; } = new StatusUpdateTracker(1);
 
+        private readonly Subject<VirtualFile> _nextIndexedFile = new Subject<VirtualFile>();
+        public IObservable<VirtualFile> IndexingObservable => _nextIndexedFile;
+
+        private readonly FileStream _vfsFileStream;
+        private readonly BinaryWriter _vfsBinaryWriter;
+
         public WorkQueue  Queue { get; }
         public bool UseExtendedHashes { get; set; }
+        public string VFSFile { get; set; }
+
+        public CompositeDisposable CompositeDisposable;
 
-        public Context(WorkQueue queue, bool extendedHashes = false)
+        public Context(WorkQueue queue, string vfsCacheFile, bool extendedHashes = false)
         {
             Queue = queue;
             UseExtendedHashes = extendedHashes;
-        }
+            VFSFile = vfsCacheFile;
+
+            CompositeDisposable = new CompositeDisposable();
 
+            _vfsFileStream = File.Open(VFSFile, FileMode.Create);
+            _vfsBinaryWriter = new BinaryWriter(_vfsFileStream, Encoding.UTF8, true);
+
+            IndexingObservable.Subscribe(AppendToVFS, () =>
+            {
+                //TODO: onComplete
```

Comment:
Don't necessarily need an on complete lambda here.  This won't actually be called, actually, unless the `.Complete()` call is made on the `Subject`.

Even, then, though, I don't think there's a specific action that needs to be taken once all the items are processed and no more are coming down the pipe?

Reply:
Also, I would consider adding an
```.ObserveOn(RxApp.TaskpoolScheduler)```
So that the work of saving to disk in `AppendToVFS` is done on some random thread, not on the same thread that fired the subject